### DISCUSSION TITLE
Stored Procedure QA Summary

### DISCUSSION THREAD

**Discussion by KingSkyLi** [TuGraph-family/tugraph-db#105] [Q&A]
本贴是存储过程的QA汇总贴，详见下方回答。
This post is a QA summary post of the stored procedure, see the answer below for details.

**Comment by KingSkyLi** [comment]
中文：
Q：如何加载存储过程或算法包？
A：加载方式有两种：

   - 第一种：通过可视化页面的插件模块，通过交互操作完成加载。
   - 第二种：通过 cypher 语句实现存储过程的加载。

   ```cypher
   CALL db.plugin.loadPlugin(plugin_type::STRING,plugin_name::STRING,plugin_content::STRING,code_type::STRING,plugin_description::STRING,read_only::BOOLEAN) :: (::VOID)
   ```
English:
Q: How to load stored procedure or algorithm package?
A: There are two loading methods:

   - The first type: through the plug-in module of the visualization page, the loading is completed through interactive operations.
   - The second type: load the stored procedure through the cypher statement.

   ```cypher
   CALL db.plugin.loadPlugin(plugin_type::STRING,plugin_name::STRING,plugin_content::STRING,code_type::STRING,plugin_description::STRING,read_only::BOOLEAN) :: (::VOID)
   ```

**Comment by KingSkyLi** [comment]
中文：
Q：如何调用或执行存储过程？
A：可以使用 cypher 进行存储过程的执行或调用。

   ```cypher
   CALL db.plugin.callPlugin(plugin_type::STRING,plugin_name::STRING,param::STRING,timeout::DOUBLE,in_process::BOOLEAN) :: (success::BOOLEAN,result::STRING)
   ```

English：
Q: How to call or execute stored procedures?
A: You can use cypher to execute or call stored procedures.

   ```cypher
   CALL db.plugin.callPlugin(plugin_type::STRING,plugin_name::STRING,param::STRING,timeout::DOUBLE,in_process::BOOLEAN) :: (success::BOOLEAN,result::STRING)
   ```

**Comment by KingSkyLi** [comment]
中文：
Q:开源内置的算法包在哪里？
A：代码地址[https://github.com/TuGraph-db/tugraph-db/tree/master/plugins](https://github.com/TuGraph-db/tugraph-db/tree/master/plugins)

English：
Q: Where is the open source built-in algorithm package?
A: Code address [https://github.com/TuGraph-db/tugraph-db/tree/master/plugins](https://github.com/TuGraph-db/tugraph-db/tree/master/plugins)

**Reply by leifchen** [reply]
内置算法包的目录是不是调整了，文档没更新？
